### PR TITLE
Use the ISAL to accelerate CRC calculation

### PR DESCRIPTION
**Pull request by hhb584520**
ISA-L library will use AVX512 to accelerate CRC calculation
more detail can be found in https://github.com/intel/isa-l/

Considering the real scene that the use of the server and the
compilation server may not be the same server, it is more 
appropriate that the relevant hardware support libraries should
be compiled into the compilation time, then select the required
library when it is actually used.

Signed-off-by: Haibin Huang <haibin.huang@intel.com>

### CODE REVIEW

File: src/overlaybd/fs/zfile/crc32/crc32c.cpp [hunk 1]
```diff
@@ -625,15 +626,43 @@ static uint32_t crc32c_dml(const uint8_t *data, size_t nbytes, uint32_t crc) {
...
+    pci_init(pacc);
+    pci_scan_bus(pacc);
+    for (dev = pacc->devices; dev; dev = dev->next) {
+        pci_fill_info(dev, PCI_FILL_IDENT | PCI_FILL_BASES);
+        if ((dev->vendor_id == 0x8086) && (dev->device_id == 0x0b25)) {
+            pci_cleanup(pacc);
+            return true;
+        }
+    }
+    std::cout << "No DSA found!" << std::endl;
```
Comment (beef9999) [inline src/overlaybd/fs/zfile/crc32/crc32c.cpp:652]: Suggest not print to stdout here. At least include alog.h and use LOG_INFO()

Comment (hhb584520) [reply src/overlaybd/fs/zfile/crc32/crc32c.cpp:652]: ok

File: src/overlaybd/fs/zfile/CMakeLists.txt [hunk 2]
```diff
@@ -2,29 +2,31 @@ file(GLOB SOURCE_ZFILE "*.cpp")
...
+add_custom_command(OUTPUT isal
+    PRE_BUILD
+    COMMAND rm -rf isa-l/
+    COMMAND git clone --recursive https://github.com/intel/isa-l.git
+    COMMAND cd isa-l && git checkout ad8dce15c6d3f0c7f3d1b486d9c649ed39223b45
+    COMMAND cd isa-l && ./autogen.sh
+    COMMAND cd isa-l && ./configure
+    COMMAND cd isa-l && make
+    COMMAND cd isa-l && sudo make install
+    VERBATIM
```
Comment (beef9999) [inline src/overlaybd/fs/zfile/CMakeLists.txt:24]: Any method to avoid make install?

Comment (hhb584520) [reply src/overlaybd/fs/zfile/CMakeLists.txt:24]: we will try it

**Review by beef9999 — APPROVED** [review]

**Review by liulanzheng — APPROVED** [review]
LGTM

**Pull request merged**